bson/timestamp.py

```diff
@@ -54,7 +55,7 @@ def __init__(self, time, inc):
         """
         if isinstance(time, datetime.datetime):
             if time.utcoffset() is not None:
-                time = time - time.utcoffset()
+                time = time - time.utcoffset()  # type: ignore
```

Comment:
Is this ignore a workaround for a mypy bug?
```
bson/timestamp.py:58: error: No overload variant of "__sub__" of "datetime" matches argument type "None"
bson/timestamp.py:58: note: Possible overload variants:
bson/timestamp.py:58: note:     def __sub__(self, timedelta) -> datetime
bson/timestamp.py:58: note:     def __sub__(self, datetime) -> timedelta
bson/timestamp.py:58: note: Right operand is of type "Optional[timedelta]"
```

Reply:
This is the same issue where the return type of a function can't be used as a "not None" guard

Reply:
I added the intermediate variable workaround